modules/Bio/EnsEMBL/Compara/RunnableDB/OrthologQM/FlagHighConfidenceOrthologs.pm

```diff
@@ -173,4 +198,58 @@ sub _write_distribution {
     }
 }
 
+sub _check_homology_counts {
+    my $self = shift;
+
+    my $homology_file = $self->param_required('homology_file');
+    my $wc_hom = $self->_lines_in_file($homology_file);
+    return (0,0,0) if $wc_hom == 0;
+
+    my $wc_goc = $self->_lines_in_file($self->param('goc_file'));
+    my $wc_wga = $self->_lines_in_file($self->param('wga_file'));
+    return ($wc_hom, $wc_goc, $wc_wga);
+}
+
+sub _lines_in_file {
+    my ( $self, $filename ) = @_;
+
+    my @wc_out = split( /\s+/, $self->get_command_output("wc -l $filename") );
+    my $wc_l   = shift @wc_out;
+    return $wc_l - 1; # account for header line
+}
+
+sub _parse_flatfile_into_hash {
+    my ($self, $filename, $filter) = @_;
+
+    my %flatfile_hash;
+    open(my $fh, '<', $filename) or die "Cannot open $filename for reading";
```

Comment:
Not all mlss_ids have GOC / WGA attached to them. Will the Runnable die in those cases ?

Reply:
I think it would have died, yes. Have addressed the issue here: https://github.com/Ensembl/ensembl-compara/pull/167/commits/575bbb3bcd62dfddcd19c63ebacc541be6e88e42